csrfguard/src/main/java/org/owasp/csrfguard/config/PropertiesConfigurationProvider.java

```diff
@@ -563,13 +568,28 @@ private void javascriptInitParamsIfNeeded() {
 				} else if (servletConfig.getServletContext().getRealPath(this.javascriptSourceFile) != null) {
 					this.javascriptTemplateCode = CsrfGuardUtils.readFileContent(servletConfig.getServletContext().getRealPath(this.javascriptSourceFile));
 				} else {
-					throw new IllegalStateException("getRealPath failed for file " + this.javascriptSourceFile);
+					try( final InputStream inputStream = getResourceStream(this.javascriptSourceFile, servletConfig)){
+						this.javascriptTemplateCode = CsrfGuardUtils.readInputStreamContent(inputStream);	
```

Comment:
If inputStream is not null then CsrfGuardUtils.readInputStreamContent will read the data from stream and assign the string into this.javascriptTemplateCode.

if inputStream is null then CsrfGuardUtils.readInputStreamContent will throw IoException which will be handled by catch block , which will further throw  IllegalStateException with our custom message.